### PR TITLE
Add achievement#show page and refactor achievement views

### PR DESCRIPTION
**Pull request by weiqingtoh**
This PR breaks up certain achievement views into partials, as there will be re-use of certain views in the manual award of achievements. Also, it implements achievement#show, which shows all the `course_users` who have obtained this particular achievement. 

Next step would be to define the abilities and controller actions for the manual award achievements.

### CODE REVIEW

File: app/views/course/achievements/index.html.slim [hunk 1]
```diff
@@ -2,11 +2,8 @@
   = new_button([current_course, :achievement]) if can?(:create, Course::Achievement.new(course: current_course))
 table.table.achievement-list.table-hover
   thead
-    tr
-      th = t('.badge')
-      th = t('.title')
-      th = t('.description')
-      th = t('.requirements')
-      th
+    = render partial: 'achievement_header', locals: { include_requirements: true }
```
Comment (lowjoel) [inline app/views/course/achievements/index.html.slim]: Don't split the partial like this. Make an achievements partial that does both, the index will just `render partial: 'achievements', local: {...}`

File: app/views/course/achievements/_achievement.html.slim [hunk 2]
```diff
@@ -1,27 +1,38 @@
-= content_tag_for(:tr, achievement,
-                  class: draft_class(achievement) + \
-                         achievement_status_class(achievement, current_course_user)) do
+table.table.achievement-list.table-hover
```
Comment (lowjoel) [inline app/views/course/achievements/_achievement.html.slim]: by convention a singular partial renders only one item, with the name of the variable being the same as the partial. Make this plural.

File: app/views/course/achievements/index.html.slim [hunk 3]
```diff
@@ -1,12 +1,5 @@
...
-    tr
-      th = t('.badge')
-      th = t('.title')
-      th = t('.description')
-      th = t('.requirements')
-      th
-  tbody
-    = render @achievements
+
+= render partial: 'achievement',
```
Comment (lowjoel) [inline app/views/course/achievements/index.html.slim]: render partial: 'achievements', ...

File: app/views/course/achievements/_achievement.html.slim [hunk 4]
```diff
@@ -1,27 +1,38 @@
...
+  thead
+    tr
+      th = t('.badge')
+      th = t('.title')
+      th = t('.description')
+      th = t('.requirements') if include_requirements
+      th
+  tbody
+    - achievements.each do |achievement|
```
Comment (lowjoel) [inline app/views/course/achievements/_achievement.html.slim]: render partial: 'achievement', collection: achievements

**Comment by weiqingtoh** [comment]
Okay updated!

File: app/views/course/achievements/_achievement.html.slim [hunk 5]
```diff
@@ -1,6 +1,6 @@
 = content_tag_for(:tr, achievement,
-                  class: draft_class(achievement) + \
-                         achievement_status_class(achievement, current_course_user)) do
+  class: draft_class(achievement) + \
```
Comment (lowjoel) [inline app/views/course/achievements/_achievement.html.slim]: Whys the spacing so strange?  Follow normal Ruby style, align them please

File: app/views/course/achievements/_achievement.html.slim [hunk 6]
```diff
@@ -10,7 +10,7 @@
 
   th
     = link_to format_inline_text(achievement.title),
-              course_achievement_path(current_course, achievement)
+      course_achievement_path(current_course, achievement)
```
Comment (lowjoel) [inline app/views/course/achievements/_achievement.html.slim]: The old alignment is correct.

File: app/views/course/achievements/_achievement.html.slim [hunk 7]
```diff
@@ -19,9 +19,10 @@
     div.description
       = format_html(achievement.description)
 
-  td = achievement.specific_conditions.map(&:title).to_sentence
+      - if include_requirements
```
Comment (lowjoel) [inline app/views/course/achievements/_achievement.html.slim]: Is it the mobile view or is all the whitespace going crazy =/

File: app/views/course/achievements/_achievement.html.slim [hunk 8]
```diff
@@ -19,9 +19,10 @@
...
+    td = achievement.specific_conditions.map(&:title).to_sentence
 
   td
     div.btn-group
       = edit_button([current_course, achievement]) if can?(:edit, achievement)
-      = delete_button([current_course, achievement]) if can?(:delete, achievement)
+      = delete_button([current_course, achievement]) if can?(:delete, achievement)
```
Comment (lowjoel) [inline app/views/course/achievements/_achievement.html.slim]: Missing trailing newline

File: app/views/course/achievements/show.html.slim [hunk 9]
```diff
@@ -0,0 +1,15 @@
+- add_breadcrumb format_inline_text(@achievement.title)
+= page_header format_inline_text(@achievement.title)
+
+= render partial: 'achievements',
+         locals: { include_requirements: false, achievements: [@achievement] }
+
+div.user-achievements
+  h2 = t('.secondary_header')
+  div.row
+    - @achievement.course_users.includes(:user).each do |course_user|
```
Comment (lowjoel) [inline app/views/course/achievements/show.html.slim]: The includes should be done in the controller.

File: app/views/course/achievements/show.html.slim [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+      div.col-xs-4.col-sm-3.col-md-2.col-lg-1
+        = div_for(course_user, class: ['text-center']) do
```
Comment (lowjoel) [inline app/views/course/achievements/show.html.slim]: You should not text center here, make a sass class and use that here instead so other people can style

File: app/views/course/achievements/show.html.slim [hunk 11]
```diff
@@ -0,0 +1,15 @@
+- add_breadcrumb format_inline_text(@achievement.title)
+= page_header format_inline_text(@achievement.title)
+
+= render partial: 'achievements',
+         locals: { include_requirements: false, achievements: [@achievement] }
+
+div.user-achievements
+  h2 = t('.secondary_header')
```
Comment (lowjoel) [inline app/views/course/achievements/show.html.slim]: I would prefer some key that evokes an idea of what does section does. Maybe users_with_badge

**Pull request merged**